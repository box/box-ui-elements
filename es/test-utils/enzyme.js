function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map