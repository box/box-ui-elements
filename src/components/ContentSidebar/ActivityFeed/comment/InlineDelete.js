/**
 * @flow
 * @file Inline Delete component
 */

import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import classNames from 'classnames';

import PlainButton from 'box-react-ui/lib/components/plain-button';
import { Flyout, Overlay } from 'box-react-ui/lib/components/flyout';
import IconTrash from 'box-react-ui/lib/icons/general/IconTrash';

import messages from '../../../messages';
import { ACTIVITY_TARGETS } from '../../../../interactionTargets';

type Props = {
    onDelete: Function,
    permissions: BoxItemPermission,
    id: string,
    intl: any,
    message: React.Node,
};

type State = {
    isConfirming?: boolean,
};

class InlineDelete extends React.Component<Props, State> {
    state = {
        isConfirming: false,
    };

    onDeleteConfirmedHandler = (): void => {
        const { id, onDelete, permissions } = this.props;
        onDelete({ id, permissions });
    };

    handleFlyoutOpen = (): void => {
        this.setState({ isConfirming: true });
    };

    handleFlyoutClose = (): void => {
        this.setState({ isConfirming: false });
    };

    render(): React.Node {
        const { intl, message } = this.props;
        return (
            <div
                className={classNames('bcs-comment-delete-container', {
                    'bcs-is-confirming': this.state.isConfirming,
                })}
            >
                <Flyout
                    onClose={this.handleFlyoutClose}
                    onOpen={this.handleFlyoutOpen}
                    position="middle-left"
                >
                    <PlainButton
                        aria-label={intl.formatMessage(messages.deleteLabel)}
                        className="bcs-comment-delete"
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE}
                    >
                        <IconTrash />
                    </PlainButton>
                    <Overlay>
                        <b>{message}</b>
                        <div>
                            <PlainButton
                                className="lnk bcs-comment-delete-yes"
                                onClick={this.onDeleteConfirmedHandler}
                                type="button"
                            >
                                <FormattedMessage
                                    {...messages.commentDeleteConfirm}
                                />
                            </PlainButton>
                            {' / '}
                            <PlainButton
                                className="lnk bcs-comment-delete-no"
                                type="button"
                            >
                                <FormattedMessage
                                    {...messages.commentDeleteCancel}
                                />
                            </PlainButton>
                        </div>
                    </Overlay>
                </Flyout>
            </div>
        );
    }
}

export { InlineDelete as InlineDeleteBase };
export default injectIntl(InlineDelete);
