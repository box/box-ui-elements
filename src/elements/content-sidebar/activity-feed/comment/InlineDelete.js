/**
 * @flow
 * @file Inline Delete component
 */

import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import classNames from 'classnames';

import PlainButton from '../../../../components/plain-button';
import { Flyout, Overlay } from '../../../../components/flyout';
import IconTrash from '../../../../icons/general/IconTrash';

import messages from '../../../common/messages';
import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';

type Props = {
    id: string,
    message: React.Node,
    onDelete: Function,
    permissions: BoxItemPermission,
} & InjectIntlProvidedProps;

type State = {
    isConfirming?: boolean,
};

class InlineDelete extends React.Component<Props, State> {
    state = {
        isConfirming: false,
    };

    onDeleteConfirmedHandler = (): void => {
        const { id, onDelete, permissions } = this.props;
        onDelete({ id, permissions });
    };

    handleFlyoutOpen = (): void => {
        this.setState({ isConfirming: true });
    };

    handleFlyoutClose = (): void => {
        this.setState({ isConfirming: false });
    };

    render(): React.Node {
        const { intl, message } = this.props;
        return (
            <div
                className={classNames('bcs-comment-delete-container', {
                    'bcs-is-confirming': this.state.isConfirming,
                })}
            >
                <Flyout onClose={this.handleFlyoutClose} onOpen={this.handleFlyoutOpen} position="middle-left">
                    <PlainButton
                        aria-label={intl.formatMessage(messages.deleteLabel)}
                        className="bcs-comment-delete"
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE}
                    >
                        <IconTrash />
                    </PlainButton>
                    <Overlay>
                        <b>{message}</b>
                        <div>
                            <PlainButton
                                className="lnk bcs-comment-delete-yes"
                                onClick={this.onDeleteConfirmedHandler}
                                type="button"
                            >
                                <FormattedMessage {...messages.commentDeleteConfirm} />
                            </PlainButton>
                            {' / '}
                            <PlainButton className="lnk bcs-comment-delete-no" type="button">
                                <FormattedMessage {...messages.commentDeleteCancel} />
                            </PlainButton>
                        </div>
                    </Overlay>
                </Flyout>
            </div>
        );
    }
}

export { InlineDelete as InlineDeleteBase };
export default injectIntl(InlineDelete);
