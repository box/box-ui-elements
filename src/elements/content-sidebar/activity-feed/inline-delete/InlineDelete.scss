.be {
    .bcs-inline-delete-container {
        margin-top: 3px;
        opacity: 0;
        position: absolute;
        right: 0;
        top: 0;
        z-index: 1;

        &.bcs-is-confirming {
            opacity: 1;
        }

        .bcs-inline-delete {
            vertical-align: top;
        }

        .bcs-inline-delete-no,
        .bcs-inline-delete-yes {
            margin: 0 3px;
        }

        .flyout-wrapper .overlay-wrapper .overlay {
            min-width: 120px;

            .overlay-title {
                white-space: nowrap;
            }
        }

        .flyout-wrapper .overlay-wrapper.left {
            .overlay {
                padding: 10px 15px;
                transform: translateY(-20%);
            }

            .arrow {
                top: 20%;
            }
        }
    }
}
