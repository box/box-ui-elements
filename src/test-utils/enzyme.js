/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
