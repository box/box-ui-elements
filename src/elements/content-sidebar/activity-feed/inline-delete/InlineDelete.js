/**
 * @flow
 * @file Inline Delete component
 */

import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import classNames from 'classnames';

import PlainButton from '../../../../components/plain-button';
import { Flyout, Overlay } from '../../../../components/flyout';
import IconTrash from '../../../../icons/general/IconTrash';

import messages from './messages';
import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';

import './InlineDelete.scss';

type Props = {
    id: string,
    message: React.Node,
    onDelete: Function,
    permissions: BoxItemPermission,
} & InjectIntlProvidedProps;

type State = {
    isConfirming?: boolean,
};

class InlineDelete extends React.Component<Props, State> {
    state = {
        isConfirming: false,
    };

    onDeleteConfirmedHandler = (): void => {
        const { id, onDelete, permissions } = this.props;
        onDelete({ id, permissions });
    };

    handleFlyoutOpen = (): void => {
        this.setState({ isConfirming: true });
    };

    handleFlyoutClose = (): void => {
        this.setState({ isConfirming: false });
    };

    render(): React.Node {
        const { intl, message } = this.props;
        return (
            <div
                className={classNames('bcs-inline-delete-container', {
                    'bcs-is-confirming': this.state.isConfirming,
                })}
            >
                <Flyout onClose={this.handleFlyoutClose} onOpen={this.handleFlyoutOpen} position="middle-left">
                    <PlainButton
                        aria-label={intl.formatMessage(messages.deleteLabel)}
                        className="bcs-inline-delete"
                        data-resin-target={ACTIVITY_TARGETS.INLINE_DELETE}
                    >
                        <IconTrash />
                    </PlainButton>
                    <Overlay>
                        <b>{message}</b>
                        <div>
                            <PlainButton
                                className="lnk bcs-inline-delete-yes"
                                onClick={this.onDeleteConfirmedHandler}
                                type="button"
                            >
                                <FormattedMessage {...messages.inlineDeleteConfirm} />
                            </PlainButton>
                            {' / '}
                            <PlainButton className="lnk bcs-inline-delete-no" type="button">
                                <FormattedMessage {...messages.inlineDeleteCancel} />
                            </PlainButton>
                        </div>
                    </Overlay>
                </Flyout>
            </div>
        );
    }
}

export { InlineDelete as InlineDeleteBase };
export default injectIntl(InlineDelete);
