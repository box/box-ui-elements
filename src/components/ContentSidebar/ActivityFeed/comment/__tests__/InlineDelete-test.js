import * as React from 'react';
import { shallow } from 'enzyme';

import { InlineDeleteBase as InlineDelete } from '../InlineDelete';

const intl = { formatMessage: () => {} };
const translationProps = {
    intl,
};

describe('components/ContentSidebar/ActivityFeed/comment/InlineDelete', () => {
    const id = '123';
    const render = (props = {}) =>
        shallow(
            <InlineDelete
                {...translationProps}
                id={id}
                message="test"
                onDelete={() => {}}
                {...props}
            />,
        );

    test('should correctly render comment', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });

    test('should set bcs-is-confirming class when confirm flyout is open', () => {
        const wrapper = render();
        const flyout = wrapper.find('Flyout');
        expect(wrapper.hasClass('bcs-is-confirming')).toBe(false);
        flyout.prop('onOpen')();
        wrapper.update();
        expect(wrapper.hasClass('bcs-is-confirming')).toBe(true);
        flyout.prop('onClose')();
        wrapper.update();
        expect(wrapper.hasClass('bcs-is-confirming')).toBe(false);
    });

    test('should call onDelete handler when comment deletion is confirmed', () => {
        const onDeleteSpy = jest.fn();
        const permissions = {
            can_edit: true,
            can_delete: false,
        };
        const wrapper = render({ onDelete: onDeleteSpy, permissions });

        const yesBtn = wrapper.find('.bcs-comment-delete-yes');
        yesBtn.simulate('click');

        expect(onDeleteSpy).toHaveBeenCalledWith({ id, permissions });
    });
});
