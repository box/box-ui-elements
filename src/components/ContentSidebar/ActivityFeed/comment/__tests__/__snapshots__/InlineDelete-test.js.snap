// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ContentSidebar/ActivityFeed/comment/InlineDelete should correctly render comment 1`] = `
<div
  className="bcs-comment-delete-container"
>
  <Flyout
    className=""
    closeOnClick={true}
    closeOnClickOutside={true}
    constrainToScrollParent={true}
    constrainToWindow={false}
    isVisibleByDefault={false}
    onClose={[Function]}
    onOpen={[Function]}
    portaledClasses={Array []}
    position="middle-left"
  >
    <PlainButton
      className="bcs-comment-delete"
    >
      <IconTrash />
    </PlainButton>
    <Overlay>
      <b>
        test
      </b>
      <div>
        <PlainButton
          className="lnk bcs-comment-delete-yes"
          onClick={[Function]}
          type="button"
        >
          <FormattedMessage
            defaultMessage="Yes"
            id="be.commentDeleteConfirm"
          />
        </PlainButton>
         / 
        <PlainButton
          className="lnk bcs-comment-delete-no"
          type="button"
        >
          <FormattedMessage
            defaultMessage="No"
            id="be.commentDeleteCancel"
          />
        </PlainButton>
      </div>
    </Overlay>
  </Flyout>
</div>
`;
